### DISCUSSION TITLE
Not able to create depth maps using gg.visualization.create_depth_maps_from_gempy

### DISCUSSION THREAD

**Discussion by soorajgeo** [cgre-aachen/gemgis#271] [Q&A] [answered]
Not able to create depth maps following the tutorial https://gemgis.readthedocs.io/en/latest/getting_started/tutorial/18_creating_depth_maps_from_gempy_models.html

I am able to compute the model but while creating depth maps, the following error occurs 
ValueError: Dimension of ``points`` should be 1 or 2, not 3

I am using the data from https://github.com/cgre-aachen/gemgis_data. Please help

**Comment by AlexanderJuestel** [comment]
@soorajgeo, thanks for your question! 

Could you send us the data to recreate the model and check what is going on?

Cheers 
Alex

**Reply by soorajgeo** [reply]
The data is from the repository https://github.com/cgre-aachen/gemgis_data/tree/main/data/example01_planar_dipping_layers. The same error happens if I run the other notebooks in the repository wherever I have to create depth maps.

**Reply by soorajgeo** [reply]
@AlexanderJuestel : Any solutions?

**Reply by AlexanderJuestel** [reply]
@soorajgeo, sorry for the late reply. Which version of GemGIS are you using?

**Reply by soorajgeo** [reply]
Sorry for my late reply too... I am using GemGIS 1.0.9 and my Gempy version is 2023.1.0

**Reply by AlexanderJuestel** [reply]
@soorajgeo, using the model and GemGIS 1.0.9, I cannot reproduce the error. Could you send some screenshots maybe?

![image](https://user-images.githubusercontent.com/45469915/227773082-85baae23-0c71-4547-a870-6063d8519ea6.png)

**Reply by soorajgeo** [reply]
@AlexanderJuestel : This is the screenshot of the error
![depth_map_error](https://user-images.githubusercontent.com/51475605/227779516-c14d2835-f538-4a72-9b07-40b43e7a5bb0.JPG)

**Reply by AlexanderJuestel** [reply]
@soorajgeo, which version of PyVista are you using?

**Reply by soorajgeo** [reply]
I am using pyvista 0.38.4

**Reply by AlexanderJuestel** [reply]
Alright, I am using 0.38.2 right now. I will upgrade to the lasted version and see if I can reproduce the error

**Reply by soorajgeo** [reply]
Ok. Thanks

**Reply by AlexanderJuestel** [reply]
I cannot reproduce it. Does the model appear in the built-in GemPy 3D plotting? `gpv = gp.plot_3d(geo_model, 
                 image=False, 
                 show_topography=True,
                 plotter_type='basic', 
                 notebook=True, 
                 show_lith=True,
                 show_boundaries=True)`

Can you also call `geo_model.surfaces.df` and see if vertices and edges have been calculated?

![image](https://user-images.githubusercontent.com/45469915/227780999-19a6a99d-5c86-4466-b721-85d9fb1d1190.png)

**Reply by soorajgeo** [reply]
Yes, I am using the built-in Gempy 3D plotting. 
![surfaces](https://user-images.githubusercontent.com/51475605/227781501-310c7528-01da-43ea-958e-1cb326bee987.JPG)

**Reply by AlexanderJuestel** [reply]
That is weird!

**Reply by AlexanderJuestel** [reply]
I am currently out of ideas. What GemPy Version are you using? You wrote 2023.1.0 but that is actually not a version. The latest is 2.2.12. 

On the other hand, it cannot be a GemPy Error but a PyVista Error....

**Reply by soorajgeo** [reply]
@AlexanderJuestel : I am using the data from https://github.com/cgre-aachen/gemgis_data and the notebook is https://github.com/cgre-aachen/gemgis_data/blob/main/notebooks/02_planar_dipping_layers/example01_planar_dipping_layers.ipynb

**Reply by soorajgeo** [reply]
I got some error when I was using 2.2.12, but then I cloned one of the branches from the repo and installed it. After that the error disappeared. I cant remember what exactly was that error. I have another environment where I have installed the gempy 2.2.12. I have to install GemGIS in that. Let me try that...

**Reply by soorajgeo** [reply]
Ok. I installed gemGIS in the environment where I had gempy 2.2.12, but then while I try to import gempy, I got the following error.
Not subsurface compatibility available
WARNING (theano.tensor.blas): Using NumPy C-API based implementation for BLAS functions.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\soora\miniconda3\envs\gempy\lib\site-packages\gempy\__init__.py", line 25, in <module>
    from gempy.gempy_api import *
  File "C:\Users\soora\miniconda3\envs\gempy\lib\site-packages\gempy\gempy_api.py", line 17, in <module>
    from gempy.core.model import Project
  File "C:\Users\soora\miniconda3\envs\gempy\lib\site-packages\gempy\core\model.py", line 20, in <module>
    from gempy.plot.decorators import *
  File "C:\Users\soora\miniconda3\envs\gempy\lib\site-packages\gempy\plot\decorators.py", line 2, in <module>
    from gempy.plot.vista import GemPyToVista
  File "C:\Users\soora\miniconda3\envs\gempy\lib\site-packages\gempy\plot\vista.py", line 36, in <module>
    from pyvista.plotting import parse_color
ImportError: cannot import name 'parse_color' from 'pyvista.plotting' (C:\Users\soora\miniconda3\envs\gempy\lib\site-packages\pyvista\plotting\__init__.py)

**Reply by soorajgeo** [reply]
@AlexanderJuestel: I created another environment with python==3.9.13.  Followed the installation instructions from [gemgis installation page](https://gemgis.readthedocs.io/en/latest/getting_started/installation.html ) pip installed gemgis. conda installed aesara and pip installed gempy. After that I downgraded the pyvista to 0.37.0. Now I can create the depth maps... Thanks for the support

**Comment by soorajgeo** [comment, marked as answer]
Created another environment with python==3.9.13. Followed the installation instructions from [gemgis installation page](https://gemgis.readthedocs.io/en/latest/getting_started/installation.html) pip installed gemgis. conda installed aesara and pip installed gempy. Then downgraded pyvista to 0.37.0.
